src/main/java/com/starrocks/connector/flink/table/source/StarRocksDynamicTableSource.java

```diff
@@ -0,0 +1,140 @@
+        }
+
+        Map<String, ColunmRichInfo> columnMap = StarRocksSourceCommonFunc.genColumnMap(flinkSchema);
+        List<ColunmRichInfo> colunmRichInfos = StarRocksSourceCommonFunc.genColunmRichInfo(columnMap);
+        SelectColumn[] selectColumns = StarRocksSourceCommonFunc.genSelectedColumns(columnMap, this.options, colunmRichInfos);
+
+        StarRocksDynamicLookupFunction tableFunction = new StarRocksDynamicLookupFunction(this.options, filerRichInfo, colunmRichInfos, selectColumns);
+        return TableFunctionProvider.of(tableFunction);
+    }
+
+    @Override
+    public DynamicTableSource copy() {
+        return new StarRocksDynamicTableSource(this.options, this.flinkSchema, this.pushDownHolder);
+    }
+
+    @Override
+    public String asSummaryString() {
+        return "StarRocks Table Source";
+    }
+
+    @Override
+    public boolean supportsNestedProjection() {
+        return false;
+    }
+
+    @Override
+    public void applyProjection(int[][] projectedFields) {
+        // if columns = "*", this func will not be called, so 'selectColumns' will be null
+        int[] curProjectedFields = Arrays.stream(projectedFields).mapToInt(value -> value[0]).toArray();
+        if (curProjectedFields.length == 0 ) {
+            this.pushDownHolder.setQueryType(StarRocksSourceQueryType.QueryCount);
+            return;
+        }
+        this.pushDownHolder.setQueryType(StarRocksSourceQueryType.QuerySomeColumns);
+
+        ArrayList<String> columnList = new ArrayList<>();
+        ArrayList<SelectColumn> selectColumns = new ArrayList<SelectColumn>(); 
+        for (int index : curProjectedFields) {
+            String columnName = flinkSchema.getFieldName(index).get();
+            columnList.add(columnName);
+            selectColumns.add(new SelectColumn(columnName, index));
+        }
+        String columns = String.join(", ", columnList);
+        this.pushDownHolder.setColumns(columns);
+        this.pushDownHolder.setSelectColumns(selectColumns.toArray(new SelectColumn[selectColumns.size()]));
+    }
+
+    @Override
+    public Result applyFilters(List<ResolvedExpression> filtersExpressions) {
+        List<String> filters = new ArrayList<>();
+        StarRocksExpressionExtractor extractor = new StarRocksExpressionExtractor();
+        for (ResolvedExpression expression : filtersExpressions) {
+            if (expression.getOutputDataType().equals(DataTypes.BOOLEAN()) && expression.getChildren().size() == 0) {
+                filters.add(expression.accept(extractor) + " = true");
+                continue;
+            }
+            String str = expression.accept(extractor);
+            if (str == null) {
+                continue;
+            }
+            filters.add(str);
+        }
+        Optional<String> filter = Optional.of(String.join(" and ", filters));
+        this.pushDownHolder.setFilter(filter.get());
+        return Result.of(filtersExpressions, new ArrayList<>());
```

Comment:
filters that can't be accepted here like 'like xxx' 'in (xxx, xxx)' 'between' etc... should all be returned.